src/braket/experimental/autoqasm/api.py

```diff
@@ -379,7 +389,7 @@ def _clone_function(f_source: Callable) -> Callable:
         copy.copy(f_source.__globals__),
         copy.deepcopy(f_source.__name__),
         copy.deepcopy(f_source.__defaults__),
-        copy.deepcopy(f_source.__closure__),
+        copy.copy(f_source.__closure__),
```

Comment:
Why change this line? Is it not a deep object, or are we trying to pick up some shared state?

Reply:
The new tests with the recursive functions throw an error during `deepcopy` of `__closure__`: `TypeError: cannot pickle 'cell' object`. Switching this to `copy` fixes that error and doesn't break any other tests. But I haven't investigated how this might affect other potential scenarios that aren't (yet) covered by tests.

Reply:
okay, weird. also intriguing that deepcopy relies on pickling

Reply:
I think it should be ok to handle later. Sounds like `__closure__` affects captured variables ([from here](https://www.tutorialspoint.com/closure-magic-function-in-python#:~:text=In%20Python%2C%20function%20objects%20have,object%20represents%20a%20captured%20variable.)). I feel like capturing values from the containing function should be disallowed, anyway

Reply:
Yeah, the fact that this only happens in pytest indicates that it was trying to pickle/copy some kind of state from pytest, which doesn't seem like behavior that we would necessarily want.